### PR TITLE
Add documentation on CI system

### LINKED ISSUE #78: Developer documentation: CI
Add docs detailing the CI system in full 

Key points to note:
- Jenkins structure
- CI file layout
- Adding tests
- Adding configurations
- Getting custom pipeline
- Getting access

### PR DESCRIPTION
**Pull request by oerc0122**
Add initial documentation on the CI systems

Fixes #78

### CODE REVIEW

File: docs/developer/CI.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+[CASTEP Default Tests (Jenkins)](#default-builds)
+: A set of tests run whenever default or a release branch is merged to.
+
+The main CASTEP continuous integration (CI) system is a
+[Jenkins](https://www.jenkins.io/) instance run on the
+[ANVIL](https://anvil.softeng-support.ac.uk/) service.
+
+In order to request access to the ANVIL service to check the status of testing
+please [contact here]()
```
Comment (oerc0122) [inline docs/developer/CI.md]: Needs appropriate contact link?

File: docs/developer/CI.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+download.
+
+Currently the artefacts that are preserved are:
+
+- Coverage reports
+- Linting reports
+- Failed test data
+- FORD Documentation
+
+### Reports
```
Comment (krefson) [inline docs/developer/CI.md]: A picture is worth 1000 words. Can we include a screenshot with some annotations to point out/explain the components including "doughnut" and the   stage view.

Actually, some more overall comments might as well go here. The document is currently missing some essentials.

1. The URL of the CASTEP Jenkins instance
2. Basic explanation of the web page view and definitions. What is the main entry view called? What do the colours mean?
3. Text and drop down Navigation links below the main Jenkins banner on the click-through page - best as an annotation on a page view screenshot,
4. Sidebar links - need a note (on the screenshot) that the contents are context-dependent.
5. Where to find Linter views.

And there really needs to be a basic "howto" on how to examine the logs to find out why a compile failed, find out which test(s) failed.  "What do I do if my PR fails the bitbucket pipeline" and "What do i do if my PR fails the Anvil pipeline?"

File: docs/developer/CI.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+##### Test Coverage Report
+
+Whenever a `coverage` build is run (i.e. with either the `ifort` or `gfortran`
+compiler and `opt=coverage`), Jenkins will accumulate the coverage reports and
+make them accessible in the side-bar.
+
+!!! failure "Known issue"
+
+    Due to current security settings on ANVIL the `ifort` reports are not
+    viewable through this view and must be downloaded to be viewed.
```
Comment (krefson) [inline docs/developer/CI.md:195]: Can we please explain how?  This is counterintuitive in the extreme.  I only just discovered how to do this myself a couple of days ago.

File: docs/developer/CI.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+```
+However, these are subject to change.
+
+!!! note
+
+    All `default` or manual builds will also run a test to ensure that the `dist`
+    target of the CASTEP build system matches the files expected relative to the
+    `git` information. This is to ensure that any new files which have been added to
+    the `git` tracking which are not accounted for in `dist` will be detected until
+    accounted for or explicitly excluded.
```
Comment (krefson) [inline docs/developer/CI.md]: Unnecessarily over-wordy.  All it needs to say is
"Default" and "manual" builds also include a test of the correctness and completeness of the source distribution GNU make and CMake "dist" targets.

**Review by jryates — APPROVED** [review]

**Pull request merged**